### DISCUSSION TITLE
Nav-tab form submission

### DISCUSSION THREAD

**Discussion by TheDigitalGuru** [twbs/bootstrap#38570] [Q&A]
I have a navtab that "contact me" that has a form on it. Upon submitting the form, if there are errors or there was a successful form submission, I have a div to display this message. However, on page reload, after form submission, I am then brought back to my home tab. I want to return back to my form tab so that a success or error message displays. Everything works except I'm brought back home, of which I do not want to be. Can anybody help me resolve this? My nav-tabs are buttons, not unorder list list items and not links otherwise. I am not clicking on these links to submit form so don't know that this matters, but thought I'd include this information. I am about 60 hours into searching for this answer and would love help with this. I'm sure I'll have to include a header in my php form, but don't know that that is the case. I can't even create an a tag link to accomplish this; although an a tag is not neccessary. I just tried this trouble shooting and had no luck. I just want to return back to that tab on form submission to see that error or success message. please help.  I'm using bootstrap 5.3

**Comment by patkon** [comment]
Without knowing your code, it's hard to find a correct answer. Basically, there are several methods for your request:

1. Process the form via AJAX, so you have no page reload
2. Or store the "active Tab" in Cookie or localStorage
3. Or use the hash to open a specific tab4. 

From the Docs (You can activate individual tabs in several ways):
https://getbootstrap.com/docs/5.3/components/navs-tabs/#via-javascript

You can check the first and second example from this post for localStorage- / hash-solution (Both variants use jQuery.):
https://stackoverflow.com/questions/18999501/how-can-i-keep-selected-bootstrap-tab-on-page-refresh
